src/openforms/contrib/zgw/clients/catalogi.py

```diff
@@ -139,17 +159,19 @@ def find_informatieobjecttypen(
         """
         _supports_filtering_valid_on = self.api_version >= (1, 2, 0)
 
-        params = {
+        params: InformatieObjectTypeListParams = {
             "catalogus": catalogus,
             "omschrijving": description,
         }
-
         # datumGeldigheid was added in v1.2.0 of the APIs & the APIs reject query string
         # parameters that are unkonwn, so we can not just optmistically send it :(
         if valid_on and _supports_filtering_valid_on:
             params["datumGeldigheid"] = valid_on.isoformat()
+        if self.allow_drafts:
```

Comment:
I'm not sure if this is an issue, but here I saved with unpublished types, then changed the feature flag to `False` and the select shows no selected options. When I save it, it does appear to still have the `Unpublished` types selected judging from the errors:

![image](https://github.com/user-attachments/assets/e37bc4dc-7e27-47fa-90f0-2b1cbca5e898)

Reply:
Yeah, bit of an odd one :thinking: 

It's definitely a design choice that remove systems being down should not break existing configuration - so clearing the value because it cannot be found is out of the question.

Perhaps if a value is set but no valid option found, displaying a warning of sorts would be wise.

Reply:
Turned it into this: 
![image](https://github.com/user-attachments/assets/eeb148c5-2317-44dd-8468-302d79f41b49)

The message takes feature flag on/off into account.